include/seqan3/range/container/dynamic_bitset.hpp

```diff
@@ -0,0 +1,1479 @@
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr dynamic_bitset & operator=(std::initializer_list<value_type> ilist) noexcept
+    {
+        assign(std::ranges::begin(ilist), std::ranges::end(ilist));
+        return *this;
+    }
+
+    /*!\brief Construction from literal.
+     * \param _lit The literal to construct the string for. May only contain '0' and '1'.
+     * \throws std::invalid_argument if any character is not '0' or '1'.
+     *
+     * The `char` literal is expected to be null-terminated (asserted in debug-mode). If it is not, the last character
+     * will be lost when copying to the instance of `dynamic_bitset`.
+     *
+     * ### Exceptions
+     *
+     * Throws std::invalid_argument if any character is not '0' or '1'.
+     */
+    template <size_t N>
+    constexpr dynamic_bitset(char const (&_lit)[N]) : dynamic_bitset{}
+    {
+        static_assert(N <= capacity_ + 1, "Length of string literal exceeds capacity of dynamic_bitset.");
+        assign(_lit);
+    }
+
+    /*!\brief Assign from literal.
+     * \param _lit The literal to assign the string from. May only contain '0' and '1'.
+     * \throws std::invalid_argument if any character is not '0' or '1'.
+     *
+     * The `char` literal is expected to be null-terminated (asserted in debug-mode). If it is not, the last character
+     * will be lost when copying to the instance of `dynamic_bitset`.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of _lit.
+     *
+     * ### Exceptions
+     *
+     * Throws std::invalid_argument if any character is not '0' or '1'.
+     */
+    template <size_t N>
+    constexpr dynamic_bitset & operator=(char const (&_lit)[N])
+    {
+        static_assert(N <= capacity_ + 1, "Length of string literal exceeds capacity of dynamic_bitset.");
+        assign(_lit);
+        return *this;
+    }
+
+    /*!\brief Assign from literal.
+     * \param _lit The literal to assign the string from. May only contain '0' and '1'.
+     * \throws std::invalid_argument if any character is not '0' or '1'.
+     *
+     * The `char` literal is expected to be null-terminated (asserted in debug-mode). If it is not, the last character
+     * will be lost when copying to the instance of `dynamic_bitset`.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of _lit.
+     *
+     * ### Exceptions
+     *
+     * Throws std::invalid_argument if any character is not '0' or '1'.
+     */
+    template <size_t N>
+    constexpr void assign(char const (&_lit)[N])
+    {
+        static_assert(N <= capacity_ + 1, "Length of string literal exceeds capacity of dynamic_bitset.");
+        assert(_lit[N - 1] == '\0');
+        uint64_t value{};
+        for (size_t i = 0; i != N - 1; ++i)
+        {
+            if (_lit[i] == '0')
+            {
+                value <<= 1;
+            }
+            else if (_lit[i] == '1')
+            {
+                if (i != 0)
```

Comment:
is this really needed?

Reply:
Mhm...maybe I can restructure it, but if I'm at pos 0 and I read 1, I can just do |= 1.
If I'm not at pos 0, I first need to shift it.

Reply:
but is the value at pos 0 before the shift not 0 anyway?

Reply:
Hah, indeed 😅